src/BenchmarkDotNet/Running/BenchmarkRunner.cs

```diff
@@ -9,21 +10,36 @@ namespace BenchmarkDotNet.Running
     public static class BenchmarkRunner
     {
         public static Summary Run<T>(IConfig config = null) =>
-            BenchmarkRunnerCore.Run(BenchmarkConverter.TypeToBenchmarks(typeof(T), config), config, ToolchainExtensions.GetToolchain);
+            BenchmarkRunnerCore.Run(BenchmarkConverter.TypeToBenchmarks(typeof(T), config), ToolchainExtensions.GetToolchain);
 
         public static Summary Run(Type type, IConfig config = null) =>
-            BenchmarkRunnerCore.Run(BenchmarkConverter.TypeToBenchmarks(type, config), config, ToolchainExtensions.GetToolchain);
+            BenchmarkRunnerCore.Run(BenchmarkConverter.TypeToBenchmarks(type, config), ToolchainExtensions.GetToolchain);
 
         public static Summary Run(Type type, MethodInfo[] methods, IConfig config = null) =>
-            BenchmarkRunnerCore.Run(BenchmarkConverter.MethodsToBenchmarks(type, methods, config), config, ToolchainExtensions.GetToolchain);
+            BenchmarkRunnerCore.Run(BenchmarkConverter.MethodsToBenchmarks(type, methods, config), ToolchainExtensions.GetToolchain);
 
-        public static Summary Run(Benchmark[] benchmarks, IConfig config) =>
-            BenchmarkRunnerCore.Run(benchmarks, config, ToolchainExtensions.GetToolchain);
+        public static Summary Run(Benchmark[] benchmarks, IConfig config)
+        {
+            var targetType = benchmarks?.FirstOrDefault()?.Target.Type;
+            return BenchmarkRunnerCore.Run(
+                new BenchmarkRunInfo(benchmarks, targetType, BenchmarkConverter.GetFullConfig(targetType, config)),
+                ToolchainExtensions.GetToolchain);
+        }
+
+        public static Summary Run(BenchmarkRunInfo benchmarks)
+        {
```

Comment:
Please use `=>` from C# 7.0 for such short methods

Reply:
No problem)